### DISCUSSION TITLE
Cycle through views like a "Slide mode"

### DISCUSSION THREAD

**Discussion by lakii22** [frangoteam/FUXA#427] [Q&A]
Hello, I'm developing a project to keep track of plastic dosing units and so far its going really well.

My question is related to the fact that I choose to make one view for each equipment that I'm monitoring because I have a bunch of information per unit. As in plastic factories we could have a lot of dosing units, I would like to have a "Slide Mode" that when activated can cycle through the views giving me the option to show it in a Television inside the factory.

Would be perfect to me if I can do it with a SCRIPT, because with this I would be able to bind it to a Switch Button, however, I didn't managed to make it yet by myself.

So is possible to do this with scripts or maybe with another method?

So this is how my "prototype one" is looking like:

![image](https://user-images.githubusercontent.com/7461434/180199477-085847cc-1d7b-4c6d-b74b-815b74fcf379.png)

I'm using OPC UA to gather the data in realtime.

Thanks in advance.

**Comment by unocelli** [comment]
Hi,
interesting feature, I think the best way would be to create from script a function to send commands to frontend

**Reply by lakii22** [reply]
Hi, 
I was thinking about it but didn't managed to do it with the knowledge that I have about FUXA.

I was hoping that someone give me a light on how to do it. Like some simple example.

**Comment by unocelli** [comment]
Hi,
I added the feature in the script, let me know if you need any explanations.

**Reply by lakii22** [reply]
First of all, I would like to thank you for the feature. 

And yes, if you could explain how it goes would be really good too.

**Reply by unocelli** [reply]
For example: 

- you define a FUXAServer Tag as global variable
- you define a script with interval 10 seconds and content:

```
var v = $getTag('Tag ID - FUXAServer as global variable' );
if (v === 0) {
	$setView('MainView');
} else if (v === 1) {
	$setView('View_1');
} else if (v === 2) {
	$setView('View_2');
}
v = ++v % 3;
$setTag('Tag ID - FUXAServer as global variable', v);
```

**Reply by lakii22** [reply]
Ok, thank you again...

This seems really good and easy to do.

I'll be inserting this into my project.

thanks.
